kahuna/public/js/query/query.js

```diff
@@ -82,7 +91,7 @@ query.controller('SearchQueryCtrl',
 
     $scope.$watchCollection(() => ctrl.filter, onValChange(filter => {
         filter.uploadedBy = filter.uploadedByMe ? ctrl.user.email : undefined;
-        $state.go('search.results', filter);
+        //$state.go('search.results', filter);
```

Comment:
This is quite a lot smoother as we aren't reloading the state on every search, we just update the searchParams stream, which is then mapped into the search results.